### ISSUE TITLE
Server claims 0.9.2 is current, but ours is 0.10.0

### ISSUE THREAD

**Issue by ilovezfs** [warner/magic-wormhole#179]
See https://github.com/Homebrew/homebrew-core/pull/14914

I'm trying to update the Homebrew formula but the test block fails with
```
iMac-TMP:homebrew-core joe$ brew test -vd magic-wormhole
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/magic-wormhole.rb
Testing magic-wormhole
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20170624-92376-1uhi4he.sb /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.0.0-p648/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/magic-wormhole.rb -vd
/usr/local/Homebrew/Library/Homebrew/test.rb (Formulary::FromPathLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/magic-wormhole.rb
==> /usr/local/Cellar/magic-wormhole/0.10.0/bin/wormhole receive 329017-homebrew-test
Warning: errors may occur unless both sides are running the same version
Server claims 0.9.2 is current, but ours is 0.10.0
Sending text message (3 Bytes)
On the other computer, please run: wormhole receive
Wormhole code is: 329017-homebrew-test

Warning: errors may occur unless both sides are running the same version
Server claims 0.9.2 is current, but ours is 0.10.0
text message sent
iMac-TMP:homebrew-core joe$
```

**Comment by warner** [comment]
Huh.. that's not supposed to happen. The client isn't supposed to complain unless the server advises a *newer* version.

Oh. I wonder if my test for "newer" is that horrible lazy kind where it just compares the version strings lexicographically, instead of splitting on dots and comparing the components numerically.

No, it looks like my test is for equality. Which is a problem because I didn't really intended to bump the advised CLI version again (to avoid printing warnings on the consoles of e.g. debian packaged versions, which aren't within the user's ability to upgrade easily).

Ok, I think it's time to remove that check. I'll spin an 0.10.1 soon that takes that out. Thanks for the catch!

**Comment by ilovezfs** [comment]
@warner out of curiosity where is the 0.9.2 string coming from? I couldn't find it.

**Comment by warner** [comment]
The server advertises it.. I launch the rendezvous server with `--advertise-version=0.9.2`.

Oh, in fact, the easiest fix is for me to just stop doing that.. don't even need to push out a new release.

Done. Could you confirm that the warning message has gone away?

**Comment by ilovezfs** [comment]
```
==> /usr/local/Cellar/magic-wormhole/0.10.0/bin/wormhole receive 601272-homebrew-test
Warning: errors may occur unless both sides are running the same version
Server claims 0.9.2 is current, but ours is 0.10.0
Sending text message (3 Bytes)
On the other computer, please run: wormhole receive
Wormhole code is: 601272-homebrew-test

Warning: errors may occur unless both sides are running the same version
Server claims 0.9.2 is current, but ours is 0.10.0
text message sent
iMac-TMP:~ joe$
```
which is running
```
  test do
    n = rand(1e6)
    pid = fork do
      exec bin/"wormhole", "send", "--code=#{n}-homebrew-test", "--text=foo"
    end
    sleep 1
    begin
      received = shell_output("#{bin}/wormhole receive #{n}-homebrew-test")
      assert_match received, "foo\n"
    ensure
      Process.wait(pid)
    end
  end
```

**Comment by warner** [comment]
Rats, I was wrong about how the server's welcome message is constructed: the `--advertise-version=` parameter can override the Versioneer-detected version, but omitting the parameter doesn't disable the advertisement.

I'll spin an 0.10.1 which makes the server behave better, and then update the server, which should fix the problem for 0.10.0 too.

**Comment by warner** [comment]
I just released 0.10.1, and upgraded+bounced the server. I'm pretty sure the problem should be gone now. @ilovezfs could you confirm?

**Comment by ilovezfs** [comment]
```
iMac-TMP:homebrew-core joe$ brew test -vd magic-wormhole
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/magic-wormhole.rb
Testing magic-wormhole
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20170626-63421-14brroj.sb /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.0.0-p648/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/magic-wormhole.rb -vd
/usr/local/Homebrew/Library/Homebrew/test.rb (Formulary::FromPathLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/magic-wormhole.rb
==> /usr/local/Cellar/magic-wormhole/0.10.0/bin/wormhole receive 245085-homebrew-test
Sending text message (3 Bytes)
On the other computer, please run: wormhole receive
Wormhole code is: 245085-homebrew-test

text message sent
```

**Comment by warner** [comment]
Ok, good. I flubbed something else, travis is complaining about the 0.10.1 release, but that test is not failing locally.. I have to track it down and spin a .2, but I don't think it should hold up your work.

**Comment by warner** [comment]
Ah, the test only failed when run on a tagged tree: when in an untagged tree, we get a "local" version string (e.g. 0.10.1+foo), and the client's version-comparison code ignores local versions.

**Comment by ilovezfs** [comment]
@warner thanks! I've shipped 0.10.0 in Homebrew.

**Comment by warner** [comment]
great, thanks!
